express/dataset.py

```diff
@@ -0,0 +1,212 @@
+from abc import abstractmethod
+import argparse
+import json
+from pathlib import Path
+
+import dask.dataframe as dd
+
+from express.component_spec import ExpressComponent, kubeflow2python_type
+from express.manifest import Manifest, Subset, Index
+
+
+class FondantDataset:
+    """Wrapper around the manifest to download and upload data into a specific framework.
+
+    Uses Dask Dataframes for the moment.
+    """
+
+    def __init__(self, manifest):
+        self.manifest = manifest
+
+    @property
+    def project_name(self):
+        return self.manifest.project_name
+
+    def _load_subset(self, name: str) -> dd.DataFrame:
+        # get subset from the manifest
+        subset = self.manifest.get_subset(name)
+        # get its location and fields
+        # TODO remove gcp prefix
+        location = "gcs://" + subset.location + ".parquet"
+        fields = list(subset.fields.keys())
+
+        df = dd.read_parquet(
+            location, columns=fields, storage_options={"project": self.project_name}
```

Comment:
oh I see you're using it here, I would just define custom `storage_options` that the user has to pass. They might differ from one provider to the other + I'm not sure if the it's required to pass the `project` if you have the right credentials

Reply:
Where would you define these?

Reply:
I think we can have the user define them once somewhere in a central config and then pass them in internally in the `FondantPipeline` class. Similar to the `metadata args` which might have to passed just as an input